packages/flutter/lib/src/widgets/widget_state.dart

```diff
@@ -9,6 +9,64 @@ import 'package:flutter/services.dart';
 // Examples can assume:
 // late BuildContext context;
 
+/// This class allows [WidgetState] enum values to be combined
+/// and resolved using [WidgetStateProperty.map].
+///
+/// {@macro flutter.widgets.WidgetStateProperty.WidgetStateMap}
+abstract class WidgetStateMapKey with WidgetStateOperators {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const WidgetStateMapKey();
+
+  /// A function used by [WidgetStateProperty.map] to check whether
+  /// each key matches the current set of states.
+  bool call(Set<WidgetState> states);
+}
+
+/// A private class, used in [WidgetStateOperators].
```

Comment:
trivial nit: this should be `//` not `///` per the style guide, because we wouldn't want to publish the documentation in its current state if we made the class public. see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-for-public-quality-private-documentation